primitives/runtime/src/generic/header.rs

```diff
@@ -122,6 +123,35 @@ where
 	}
 }
 
+impl<Number, Hash> TypeInfo for Header<Number, Hash>
```

Comment:
AFAICT the encoding/decoding/typeinfo could be derived with an attribute `#[codec(compact)]` but let's do in a follow-up.

Reply:
I think you are right, the reason I didn't do that originally was because the `#[codec(compact)]` attribute was not allowed without derived `Encode`. However that is now fixed with https://github.com/paritytech/scale-info/pull/117

Reply:
Done